### PR TITLE
Add aiohttp-apispec compatibility shims and expand migration guide

### PR DESCRIPTION
**Pull request by kulapard**
## Summary

A thorough backward-compatibility audit against **aiohttp-apispec 2.2.3** (the last stable release, webargs < 6) found 8 undocumented breaking changes on top of the already-documented nested `error.messages` — three of them **silent** (code imports and runs, but validation behaves differently). This PR restores compatibility where it is cheap and documents everything else.

### Code (compat shims)

- **`use_kwargs` / `marshal_with` aliases** for `request_schema` / `response_schema` — previously `ImportError` on code migrating from aiohttp-apispec.
- **`locations=[...]` shim** on `request_schema`: previously the kwarg was silently swallowed by `**kwargs` and validation fell back to the `json` location. Now a single-element list works (with `DeprecationWarning`); a non-list raises `TypeError`; an empty or multi-element list raises `ValueError` (webargs 8 can't parse several locations with one schema); combining explicit `location=` (even `"json"`) with `locations=` raises `ValueError`.
- **`TypeError` on unrecognized `request_schema` kwargs** — closes the silent-swallowing hole for good (typos included).
- **Built-in `flat_error_handler`** — error callback restoring the flat aiohttp-apispec 2.x validation error format (strips the webargs 8 location level from `error.messages`, responds 422). Logs an error when flattening multiple locations could overwrite same-named fields. `setup_aiohttp_apispec(app, error_callback=flat_error_handler)`.
- **`request["data"]` default `[]` → `{}`** when no schema targets the default key, matching aiohttp-apispec 2.x (the `[]` was copied from the 3.0.0b2 beta).

### Documented only (intentional webargs 8 semantics, not patched)

- Default parse location is now `json` only (webargs 5 searched `querystring` → `form` → `json`).
- Unknown request fields now raise 422 (marshmallow `RAISE` default); workaround: `Meta.unknown = ma.EXCLUDE`.
- Multiple schemas without `put_into`: merge → first-wins.
- Two schemas on one location: `RuntimeError` at decoration time.
- Internal string app keys (`_apispec_parser`, `_apispec_request_data_name`) removed; `app["swagger_dict"]` kept.

The README "Migration from aiohttp-apispec" section is rewritten into "silent" / "loud" groups with workarounds for each item (the flat-format workaround now points to the built-in `flat_error_handler`); CHANGELOG has an Unreleased entry marking the breaking bits.

### Upgrade impact for existing apigami users

Two changes are breaking for projects already on apigami (both flagged **Breaking** in CHANGELOG), so this should ship as **0.8.0**, not a patch release:

1. `request["data"]` defaults to `{}` instead of `[]` when no schema targets the default key — affects code that compares with `== []`, indexes, or returns the value as-is.
2. An unrecognized `request_schema` keyword argument now raises `TypeError` at decoration (import) time instead of being silently ignored — loud, and exposes a latent bug, but a previously "working" deploy will fail fast.

Edge case: code that already passed `locations=["querystring"]` silently parsed `json` before; it now parses `querystring` (the intended behavior) and emits a `DeprecationWarning`.

### Tests

- `tests/decorators/test_apispec_compat.py`: alias identity, `locations=` shim + `DeprecationWarning`, empty/multi-element and both-args `ValueError`, non-list `TypeError`, unknown-kwarg `TypeError`, `required=` still accepted.
- `tests/test_flat_error_handler.py`: location level stripped, multi-location merge + logged warning, non-dict messages passthrough, end-to-end 422 with flat body.
- Updated `test_match_info` for the `{}` default.
- Full suite: 193 passed; mypy and pre-commit (ruff, codespell) clean.

### CODE REVIEW

File: aiohttp_apigami/decorators/request.py [hunk 1]
```diff
@@ -115,14 +116,36 @@ async def index(request):
...
+            stacklevel=2,
+        )
+        if location != "json":
+            raise ValueError("Use either `location` or `locations`, not both")
+        if len(locations) != 1:
+            raise ValueError(
+                f"Multiple locations are not supported: {locations}. "
+                f"Use a separate @request_schema(..., location=...) decorator per location"
+            )
+        location = locations[0]
```
[the remark this answers is not in the document]
Comment (kulapard) [reply aiohttp_apigami/decorators/request.py]: Fixed in eb89cb2: `locations` must now be a `list`/`tuple` — anything else (including a bare string) raises `TypeError`, and an empty or multi-element list raises `ValueError` with an explicit message.

File: aiohttp_apigami/decorators/request.py [hunk 2]
```diff
@@ -115,14 +116,36 @@ async def index(request):
...
+    locations = kwargs.pop("locations", None)
+    if locations is not None:
+        warnings.warn(
+            "The `locations` argument is deprecated, use `location` instead",
+            DeprecationWarning,
+            stacklevel=2,
+        )
+        if location != "json":
+            raise ValueError("Use either `location` or `locations`, not both")
+        if len(locations) != 1:
```
[the remark this answers is not in the document]
Comment (kulapard) [reply aiohttp_apigami/decorators/request.py]: Fixed in eb89cb2: `location` now defaults to a sentinel, so any explicit `location` (including `location="json"`) combined with `locations=[...]` raises `ValueError`, matching the README/CHANGELOG. Covered by a parametrized test with both `form` and `json`.
